apacheconfig/wloader.py

```diff
@@ -0,0 +1,423 @@
+        config string.
+
+        Args:
+            raw_str (str): Config string to parse.
+            options (dict): Additional options to pass to the created parser.
+                Ignored if another ``parser`` is supplied.
+            parser (:class:`apacheconfig.ApacheConfigParser`): optional, to
+                re-use an existing parser. If ``None``, creates a new one.
+        Returns:
+            :class:`apacheconfig.ContentsNode` containing data parsed from
+            ``raw_str``.
+        """
+        if not parser:
+            parser = _create_apache_parser(options, start='contents')
+        return ContentsNode(parser.parse(raw_str))
+
+    def add(self, index, raw_str):
+        """Parses given string into appropriate :class:`apacheconfig.Node`
+        object, then adds to contents at specified index.
+
+        Args:
+            index (int): index of contents at which to insert the node.
+            raw_str (str): string to parse. The parser will automatically
+                determine whether it's a :class:`apacheconfig.BlockNode` or
+                :class:`apacheconfig.ItemNode`.
+
+        Returns:
+            The :class:`apacheconfig.Node` created from parsing ``raw_str``.
+        """
+        parser = _create_apache_parser({}, start='miditem')
+        raw = parser.parse(raw_str)
+        if raw[0] == "block":
+            node = BlockNode(raw)
+        else:
+            node = ItemNode(raw)
+        self._contents.insert(index, node)
+        if (index + 1 < len(self._contents) and
+           '\n' not in self._contents[index + 1].whitespace):
+            whitespace_after = self._contents[index + 1].whitespace
+            self._contents[index + 1].whitespace = '\n' + whitespace_after
+        return node
+
+    def remove(self, index):
+        """Removes node from supplied index.
+
+        Args:
+            index (int): index of node to remove from contents.
+
+        Returns:
+            The :class:`apacheconfig.Node` that was removed.
+        """
+        thing = self._contents[index]
+        del self._contents[index]
+        return thing
+
+    def __len__(self):
+        """Number of :class:`apacheconfig.Node` children."""
+        return len(self._contents)
+
+    def __iter__(self):
+        """Iterator over :class:`apacheconfig.Node` children."""
+        return iter(self._contents)
+
+    def dump(self):
+        """See base class."""
+        return ("".join([item.dump() for item in self._contents])
+                + self.whitespace)
+
+    @property
+    def ast_node_type(self):
+        """See base class. Can only return ``"contents"`` for
+        :class:`apacheconfig.ContentsNode`.
+        """
+        return self._type
+
+    @property
+    def whitespace(self):
+        """Returns trailing whitespace in contents.
+
+        For instance, the following valid "contents" configuration::
+
+            \\tkey value # comment\\n
```

Comment:
nit: may be `code-block:: something`?

Reply:
I think the trailing `::` at the end of the previous line converts the following paragraph to a monospace block, though there's no syntax highlighting. Should b fine since it's pseudo-code anyways :)